lib/active_fedora/relation/finder_methods.rb

```diff
@@ -189,17 +189,16 @@ def find_one(id, cast = nil)
     protected
 
       def load_from_fedora(id, cast)
-        raise ActiveFedora::ObjectNotFoundError if id.empty?
+        if id.empty?
+          raise ActiveFedora::ObjectNotFoundError, "No ID provided for #{klass.name}."
+        end
         resource = ActiveFedora.fedora.ldp_resource_service.build(klass, id)
-        raise_record_not_found_exception!(id) if resource.new?
+        if resource.new?
+          raise ActiveFedora::ObjectNotFoundError, "No #{klass.name} with id #{id} exists."
+        end
         class_to_load(resource, cast).allocate.init_with_resource(resource) # Triggers the find callback
       end
 
-      def raise_record_not_found_exception!(id)
```

Comment:
This method was specifically added to match the https://github.com/rails/rails/blob/385799a4a3a361ed2d27a5e72e884f1b7d82f096/activerecord/lib/active_record/relation/finder_methods.rb#L339.   This was prerequisite for implementing the `first!`, `second!`, `take!` methods.  Please revert this.